src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newElement = new Node<T>(value);
+            head = newElement;
+            tail = newElement;
+        } else {
+            Node<T> newElement = new Node<T>(value);
+            newElement.prev = tail;
+            Node<T> previousNode = tail;
```

Comment:
The variable 'previousNode' is redundant here as it is only used once. You can directly use 'tail' instead.